dali/operators/image/clahe/clahe_op.cu

```diff
@@ -0,0 +1,1190 @@
+    int blocks = div_up(N, threads * 4);  // Each thread processes 4 pixels
+    rgb_to_y_u8_nhwc_vectorized_kernel<<<blocks, threads, 0, stream>>>(in_rgb, y_plane, H, W);
+  } else {
+    int threads = 256;
+    int blocks = div_up(N, threads);
+    rgb_to_y_u8_nhwc_kernel<<<blocks, threads, 0, stream>>>(in_rgb, y_plane, H, W);
+  }
+}
+
+// -------------------------------------------------------------------------------------
+// Fused Kernel: RGB to Y + Histogram per tile (optimized)
+// Combines RGB->Y conversion with histogram computation to reduce memory round-trips
+// Each block handles one tile and builds histogram from RGB data directly
+// -------------------------------------------------------------------------------------
+__global__ void fused_rgb_to_y_hist_kernel(const uint8_t *__restrict__ rgb,
+                                           uint8_t *__restrict__ y_out, int H, int W, int tiles_x,
+                                           int tiles_y, unsigned int *__restrict__ histograms) {
+  extern __shared__ unsigned int shist[];  // 256 bins
+  const int bins = 256;
+
+  int tx = blockIdx.x;  // tile x
+  int ty = blockIdx.y;  // tile y
+  if (tx >= tiles_x || ty >= tiles_y)
+    return;
+
+  // Zero shared histogram
+  for (int i = threadIdx.x; i < bins; i += blockDim.x)
+    shist[i] = 0u;
+  __syncthreads();
+
+  // Compute tile bounds
+  int tile_w = div_up(W, tiles_x);
+  int tile_h = div_up(H, tiles_y);
+  int x0 = tx * tile_w;
+  int y0 = ty * tile_h;
+  int x1 = min(x0 + tile_w, W);
+  int y1 = min(y0 + tile_h, H);
+
+  // Loop over tile pixels - fused RGB->Y + histogram
+  int area = (x1 - x0) * (y1 - y0);
+  for (int i = threadIdx.x; i < area; i += blockDim.x) {
+    int dy = i / (x1 - x0);
+    int dx = i - dy * (x1 - x0);
+    int x = x0 + dx;
+    int y = y0 + dy;
+
+    int pixel_idx = y * W + x;
+    int rgb_idx = 3 * pixel_idx;
+
+    // RGB to LAB L* conversion (match OpenCV exactly)
+    // First convert to normalized RGB [0,1]
+    float r_val = rgb[rgb_idx + 0];
+    float g_val = rgb[rgb_idx + 1];
+    float b_val = rgb[rgb_idx + 2];
+
+    float rf = r_val / 255.0f;
+    float gf = g_val / 255.0f;
+    float bf = b_val / 255.0f;
+
+    // Apply gamma correction (sRGB to linear RGB)
+    rf = srgb_to_linear(rf);
+    gf = srgb_to_linear(gf);
+    bf = srgb_to_linear(bf);
+
+    // Convert to CIE XYZ using OpenCV's exact transformation matrix
+    // From OpenCV source: cv::COLOR_RGB2Lab
+    float x_xyz = 0.412453f * rf + 0.357580f * gf + 0.180423f * bf;
+    float y_xyz = 0.212671f * rf + 0.715160f * gf + 0.072169f * bf;
+    float z_xyz = 0.019334f * rf + 0.119193f * gf + 0.950227f * bf;
+
+    // Normalize by D65 white point (OpenCV values)
+    x_xyz = x_xyz / 0.950456f;
+    y_xyz = y_xyz / 1.000000f;
+    z_xyz = z_xyz / 1.088754f;
+
+    // Convert Y to LAB L* using OpenCV's threshold and constants
+    float fy = (y_xyz > 0.008856f) ? cbrtf(y_xyz) : (7.787f * y_xyz + 16.0f / 116.0f);
```

Comment:
You used `008856f` here but 
```
  const float delta = 6.0f / 29.0f;
  const float delta_cube = delta * delta * delta;
```
in `xyz_to_lab_f`, which is way more precise. Please use a helper function in both places which uses the accurate constant.